src/tablecloth/api/join_separate.clj

```diff
@@ -116,3 +118,46 @@
      (if (grouped? ds)       
        (process-group-data ds #(process-separate-columns % column target-columns replace-missing separator-fn drop-column?) parallel?)
        (process-separate-columns ds column target-columns replace-missing separator-fn drop-column?)))))
+
+(defn array-column->columns
+  "Converts a column of type java array into several columns,
+  one for each element of the array of all rows. The source column is dropped afterwards.
+  The function assumes that arrays in all rows have same type and length and are numeric.
+
+  `ds` Datset to operate on.
+  `src-column` The (array) column to convert
+  "
+  [ds src-column]
+  (assert (not (grouped? ds)) "Not supported on grouped datasets")
+  (let [new-ds
+        (->
+         (dtt/concat-buffers (src-column ds))
```

Comment:
Column name can be anything, not only a keyword. Should be the opposite: `(ds src-column)`